File: jdroid-android-sample/src/main/java/com/jdroid/android/sample/ui/service/SampleServiceCommand4.java
package com.jdroid.android.sample.ui.service;

import android.content.Context;
import android.os.Bundle;

import com.jdroid.java.concurrent.ExecutorUtils;

import java.util.concurrent.TimeUnit;

import androidx.annotation.Nullable;

public class SampleServiceCommand4 extends AbstractSampleServiceCommand {

	@Override
	protected boolean execute(Context context, Bundle bundle) {
		ExecutorUtils.INSTANCE.sleep(20, TimeUnit.SECONDS);
		return false;
	}

	@Nullable
	@Override
	protected Long getMinimumTimeBetweenExecutions() {
		return TimeUnit.MINUTES.toMillis(1);
	}
}

File: jdroid-android-sample/src/main/java/com/jdroid/android/sample/ui/service/SampleServiceCommand4.kt
package com.jdroid.android.sample.ui.service

import android.content.Context
import android.os.Bundle

import com.jdroid.java.concurrent.ExecutorUtils

import java.util.concurrent.TimeUnit

class SampleServiceCommand4 : AbstractSampleServiceCommand() {

    override fun execute(context: Context, bundle: Bundle): Boolean {
        ExecutorUtils.sleep(20, TimeUnit.SECONDS)
        return false
    }

    override fun getMinimumTimeBetweenExecutions(): Long? {
        return TimeUnit.MINUTES.toMillis(1)
    }
}
